include/minstrobe.hpp

```diff
@@ -0,0 +1,506 @@
+    * \param[in] window_min  The lower offset for the position of the next window from the previous one.
+    * \param[in] window_max  The upper offset for the position of the next window from the previous one.
+    *
+    * \details
+    *
+    * Looks at the number of values per window with two iterators. First iterator adds the next value in the tuple as
+    * the first strobe. The second iterator adds the minimum value of the window to the second position of the tuple.
+    *
+    */
+    basic_iterator(urng1_iterator_t urng1_iterator,
+                   urng1_sentinel_t urng1_sentinel,
+                   size_t window_min,
+                   size_t window_max) :
+        urng1_iterator{std::move(urng1_iterator)},
+        urng1_sentinel{std::move(urng1_sentinel)}
+    {
+        window_first(window_min, window_max);
+    }
+    //!\}
+
+    //!\anchor basic_iterator_comparison_minstrobe
+    //!\name Comparison operators
+    //!\{
+
+    //!\brief Compare to another basic_iterator.
+    friend bool operator==(basic_iterator const & lhs, basic_iterator const & rhs)
+    {
+        return (lhs.urng1_iterator == rhs.urng1_iterator);
+    }
+
+    //!\brief Compare to another basic_iterator.
+    friend bool operator!=(basic_iterator const & lhs, basic_iterator const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to the sentinel of the minstrobe_view.
+    friend bool operator==(basic_iterator const & lhs, sentinel const &)
+    {
+        return lhs.urng1_iterator == lhs.urng1_sentinel;
+    }
+
+    //!\brief Compare to the sentinel of the minstrobe_view.
+    friend bool operator==(sentinel const & lhs, basic_iterator const & rhs)
+    {
+        return rhs == lhs;
+    }
+
+    //!\brief Compare to the sentinel of the minstrobe_view.
+    friend bool operator!=(sentinel const & lhs, basic_iterator const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to the sentinel of the minstrobe_view.
+    friend bool operator!=(basic_iterator const & lhs, sentinel const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+
+    //!\brief Pre-increment.
+    basic_iterator & operator++() noexcept
+    {
+        next_minstrobe();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    basic_iterator operator++(int) noexcept
+    {
+        basic_iterator tmp{*this};
+        next_minstrobe();
+        return tmp;
+    }
+
+    //!\brief Return the minstrobe.
+    value_type operator*() const noexcept
+    {
+        return minstrobe_value;
+    }
+
+private:
+    //!\brief The minstrobe value.
+    value_type minstrobe_value{};
+
+    //!\brief The offset relative to the beginning of the window where the minstrobe value is found.
+    size_t minstrobe_position_offset{};
+
+    //!\brief Iterator to the right most value of the window.
+    urng1_iterator_t urng1_iterator{};
+    //!\brief Iterator to the first value of minstrobe.
+    urng1_iterator_t first_iterator{};
```

Comment:
Those variable names are not that telling and don`t fit together, maybe name the other second iterator?

Reply:
Honestly I didn't name it second because it was defined first and it's the urng1 iterator type but you are right. (: